<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>MuSiC2: cell type deconvolution for multi-condition bulk RNA-seq data • MuSiC</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../bootstrap-toc.css">
<script src="../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><meta property="og:title" content="MuSiC2: cell type deconvolution for multi-condition bulk RNA-seq data">
<meta property="og:description" content="MuSiC">
<meta property="og:image" content="http://xuramw.github.io/MuSiC/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">MuSiC</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../../articles/MuSiC.html">Tutorial</a>
</li>
<li>
  <a href="../../articles/pages/MuSiC2.html">MuSiC2</a>
</li>
<li>
  <a href="../../reference/index.html">Documentation</a>
</li>
<li>
  <a href="../../articles/pages/data.html">Data</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/xuranw/MuSiC">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>MuSiC2: cell type deconvolution for multi-condition bulk RNA-seq data</h1>
                        <h4 class="author">Jiaxin Fan</h4>
            <address class="author_afil">
      Food and Drug Adminstration, MD, USA<br><h4 class="author">Xuran Wang</h4>
            <address class="author_afil">
      SEAVER Autism Center, ISMMS, New York, USA<br><small class="dont-index">Source: <a href="https://github.com/xuranw/MuSiC/blob/master/../../../Single%20cell/Rpackage/MuSiC-master_v1.0.0/vignettes/pages/MuSiC2.Rmd"><code>../../../Single cell/Rpackage/MuSiC-master_v1.0.0/vignettes/pages/MuSiC2.Rmd</code></a></small>
      <div class="hidden name"><code>MuSiC2.Rmd</code></div>

    </address>
</address>
</div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>This vignette provides a walk through tutorial on how to use <code>MuSiC2</code> to estimate cell type proportions for bulk RNA-seq data using scRNA-seq data as reference when the bulk and scRNA-seq data are generated from samples with multiple clinical conditions. The codes of MuSiC2 can also be found at this <a href="https://github.com/Jiaxin-Fan/MuSiC2">GitHub Repo</a>.</p>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>If you have never used <code>MuSiC</code> before, please install this integrated package for both <code>MuSiC</code> and <code>MuSiC2</code>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># install devtools if necessary</span>
<span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">'devtools'</span><span class="op">)</span>

<span class="co"># install the MuSiC package</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org/reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">'xuranw/MuSiC'</span><span class="op">)</span>

<span class="co"># load</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/xuranw/MuSiC">MuSiC</a></span><span class="op">)</span></code></pre></div>
<p>If you have installed <code>MuSiC</code> version 0.2.0 or lower, a separate installation of <code>MuSiC2</code> uses following:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># install devtools if necessary</span>
<span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="st">"devtools"</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">'devtools'</span><span class="op">)</span>
<span class="op">}</span>
<span class="co"># install the MuSiC2 package</span>
<span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="st">"MuSiC2"</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org/reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">'Jiaxin-Fan/MuSiC2'</span><span class="op">)</span>
<span class="op">}</span>
<span class="co"># load</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">MuSiC2</span><span class="op">)</span></code></pre></div>
</div>
<div id="data" class="section level2">
<h2 class="hasAnchor">
<a href="#data" class="anchor"></a>Data</h2>
<p>Two types of input data:</p>
<ul>
<li><p>Bulk RNA sequencing expression data collected from samples with 2 different clincial conditions, e.g., healthy and diseased. These are the data we want to deconvolve.</p></li>
<li><p>Single-cell RNA sequencing (scRNA-seq) expression data collected from samples with single condition, e.g., healthy. The cell types of scRNA-seq are pre-determined. These serve as the reference for estimating cell type proportions of the bulk data.</p></li>
</ul>
<p><span style="color:gray"> Both datasets should be in the form of <code>ExpressionSet</code>. The details of constructing <code>ExpressionSet</code> can be found on <a href="https://urldefense.proofpoint.com/v2/url?u=https-3A__www.bioconductor.org_packages_release_bioc_vignettes_Biobase_inst_doc_ExpressionSetIntroduction.pdf&amp;d=DwIGAw&amp;c=shNJtf5dKgNcPZ6Yh64b-ALLUrcfR-4CCQkZVKC8w3o&amp;r=MaqomsZfe9g2L5m-x4MLq3LavQ4CbKBHQVTQc6BzIag&amp;m=211Af93r1vESTP_xQqMKpdP9Da_C_6apfaBMhkLFY6v4fAHoR_bpFDzmB_Sa6lc-&amp;s=FUd2242mcelYcjm3WIoFDADhnbM8aPSIzIqJaRDDqi8&amp;e=">this page</a>.</span></p>
<blockquote>
<p><strong><em>UPDATE:</em></strong> Per users’ requests, we have updated <code>MuSiC2</code> functions (version 1.0.0) and <code>SingleCellExperiment</code> objects are used to handle single cell references, where sparse matrices are compatible as read counts. The details of constructing <code>SingleCellExperiment</code> objects can be found on <a href="https://bioconductor.org/packages/devel/bioc/vignettes/SingleCellExperiment/inst/doc/intro.html">this page</a>.</p>
<p>Datasets described in the table above are in <code>SingleCellExperiment</code> (single cell references) or <code>ExpressionSet</code> (bulk). They are available at the <a href="data.html">data download page</a>.</p>
</blockquote>
</div>
</div>
<div id="music2-deconvolution" class="section level1">
<h1 class="hasAnchor">
<a href="#music2-deconvolution" class="anchor"></a>MuSiC2 Deconvolution</h1>
<p>MuSiC2 is an iterative algorithm aiming to improve cell type deconvolution for bulk RNA-seq data when the bulk data and scRNA-seq reference are generated from samples with different clinical conditions. The key idea of MuSiC2 is that when the bulk and single-cell samples are from different clinical conditions, the majority of genes shall still have similar cell-type-specific gene expression patterns between conditions. By removing genes with cell-type-specific differential expression (DE) between conditions from the single-cell reference, MuSiC2 can refine the reference gene list and yield more accurate cell type proportion estimates.</p>
<div id="two-step-iterative-deconvolution" class="section level2">
<h2 class="hasAnchor">
<a href="#two-step-iterative-deconvolution" class="anchor"></a>Two-step iterative deconvolution</h2>
<p>To deconvolve bulk RNA-seq samples from both Healthy and Diseased conditions and use scRNA-seq data generated only from the Healthy condition as the reference, MuSiC2 iterates over 2 steps:</p>
<ul>
<li><p>Step 1: we use MuSiC <span class="citation">Wang et al. (2019)</span> to infer the cell type proportions of the bulk samples under both conditions by borrowing information from the scRNA-seq data;</p></li>
<li><p>Step 2: for samples within each condition, we deconvolve the bulk-level expression over the cell type proportion estimates obtained in Step 1 to infer the cell-type-specific mean expression for each gene and identify cell-type-specific DE genes between conditions. By removing genes with cell-type-specific DE from the scRNA-seq data, we can update the cell type proportion estimates in Step 1 for bulk samples generated under the Diseased condition.</p></li>
</ul>
<p>By alternating between cell type deconvolution (Step 1) and cell-type-specific DE gene detection and removal (Step 2), <code>MuSiC2</code> gradually refines the list of “stable” genes retained in the scRNA-seq reference. As a result, it improves the cell type proportion estimation for the diseased samples. An overview of <code>MuSiC2</code> is in the Figure below. <img src="../images/MuSiC2.jpg"></p>
</div>
</div>
<div id="sample-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#sample-analysis" class="anchor"></a>Sample Analysis</h1>
<p>For illustration purpose, in this tutorial, we deconvolved the benchmark bulk RNA-seq data, which contain raw RNA-seq read counts and sample annotation data for 100 healthy and 100 diseased (i.e., Type 2 diabetes (T2D)) samples simulated based on pancreatic islets scRNA-seq RNA-seq data from <span class="citation">Segerstolpe et al. (2016)</span>. The procedure for generating the benchmark dataset can be found in the Methods session of the <code>MuSiC2</code> manuscript. We deconvolved the benchmark bulk RNA-seq data using scRNA-seq data generated from 6 healthy subjects by <span class="citation">Segerstolpe et al. (2016)</span>. Both datasets can be found on <a href="data.html">this page</a>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Bulk RNA-seq data</span>
<span class="va">benchmark.eset</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="st">"https://xuranw.github.io/MuSiC/data/bulk-eset.rds"</span><span class="op">)</span>
<span class="va">benchmark.eset</span>
<span class="co">## ExpressionSet (storageMode: lockedEnvironment)</span>
<span class="co">## assayData: 10000 features, 200 samples </span>
<span class="co">##   element names: exprs </span>
<span class="co">## protocolData: none</span>
<span class="co">## phenoData</span>
<span class="co">##   sampleNames: 1 10 ... 200 (200 total)</span>
<span class="co">##   varLabels: sampleID group</span>
<span class="co">##   varMetadata: labelDescription</span>
<span class="co">## featureData: none</span>
<span class="co">## experimentData: use 'experimentData(object)'</span>
<span class="co">## Annotation:</span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># clinical conditions</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">benchmark.eset</span><span class="op">$</span><span class="va">group</span><span class="op">)</span>
<span class="co">## </span>
<span class="co">## healthy     t2d </span>
<span class="co">##     100     100</span>

<span class="va">bulk.control.mtx</span> <span class="op">=</span> <span class="fu">exprs</span><span class="op">(</span><span class="va">benchmark.eset</span><span class="op">)</span><span class="op">[</span>, <span class="va">benchmark.eset</span><span class="op">$</span><span class="va">group</span> <span class="op">==</span> <span class="st">'healthy'</span><span class="op">]</span>
<span class="va">bulk.case.mtx</span> <span class="op">=</span> <span class="fu">exprs</span><span class="op">(</span><span class="va">benchmark.eset</span><span class="op">)</span><span class="op">[</span>, <span class="va">benchmark.eset</span><span class="op">$</span><span class="va">group</span> <span class="op">==</span> <span class="st">'t2d'</span><span class="op">]</span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## scRNA-seq data</span>
<span class="va">seger.sce</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="st">"https://xuranw.github.io/MuSiC/data/EMATBsce_healthy.rds"</span><span class="op">)</span>
<span class="va">seger.sce</span>

<span class="co">#class: SingleCellExperiment </span>
<span class="co">#dim: 25453 1097 </span>
<span class="co">#metadata(0):</span>
<span class="co">#assays(1): counts</span>
<span class="co">#rownames(25453): SGIP1 AZIN2 ... KIR2DL2 KIR2DS3</span>
<span class="co">#rowData names(1): gene.name</span>
<span class="co">#colnames(1097): AZ_A10 AZ_A11 ... HP1509101_P8 HP1509101_P9</span>
<span class="co">#colData names(4): sampleID SubjectName cellTypeID cellType</span>
<span class="co">#reducedDimNames(0):</span>
<span class="co">#mainExpName: NULL</span>
<span class="co">#altExpNames(0):</span></code></pre></div>
<div id="cell-type-deconvolution" class="section level2">
<h2 class="hasAnchor">
<a href="#cell-type-deconvolution" class="anchor"></a>Cell Type Deconvolution</h2>
<p>The cell type proportions are estimated by the function <code>music2_prop</code>. The essential inputs are:</p>
<ul>
<li>
<code>bulk.eset</code>: ExpressionSet of bulk data;</li>
<li>
<code>sc.eset</code>: ExpressionSet of single cell data;</li>
<li>
<code>condition</code>: character, the phenoData of bulk dataset used for indicating clinical conditions;</li>
<li>
<code>control</code>: character, the clinical condition of bulk samples that is the same as the clinical condition of the single cell samples;</li>
<li>
<code>case</code>: character, the clinical condition of bulk samples that is different from the clinical condition of the single cell samples;</li>
<li>
<code>clusters</code>: character, the phenoData from single cell dataset used as clusters;</li>
<li>
<code>samples</code>: character, the phenoData from single cell dataset used as samples;</li>
<li>
<code>select.ct</code>: vector of cell types. Default is <code>NULL</code>, which uses all cell types provided in the single-cell data;</li>
<li>
<code>n_resample</code>: numeric, number of resamples used for detecting cell-type-specific DE genes. Default is 20;</li>
<li>
<code>sample_prop</code>: numeric, proportion of samples to be randomly sampled without replacement under each clinical condition for each resample. Default is 0.5;</li>
<li>
<code>cutoff_c</code>: numeric, cutoff on the upper quantile of <span class="math inline">\(T_g^k\)</span> statistics for detecting cell-type-specific DE genes for common cell types (i.e., cell type proportion <span class="math inline">\(\ge\)</span> 0.1). Default is 0.05;</li>
<li>
<code>cutoff_r</code>: numeric, cutoff on the upper quantile of <span class="math inline">\(T_g^k\)</span> statistics for detecting cell-type-specific DE genes for rare cell types (i.e., cell type proportion &lt; 0.1). Default is 0.01;</li>
</ul>
<p>The output of <code>music2_prop</code> is a list with elements:</p>
<ul>
<li>
<code>Est.prop</code>: matrix, cell type proportion estimates;</li>
<li>
<code>convergence</code>: logical, whether MuSiC2 converged or not;</li>
<li>
<code>n.iter</code>: numeric, number of iterations;</li>
<li>
<code>DE.genes</code>: vector, cell-type-specific DE genes being removed;</li>
</ul>
<p>For illustration purpose, we constrained our analysis on 6 well-studied cell types: acinar, alpha, beta, delta, ductal and gamma. Figure 2 below showed the estimated cell type proportion of MuSiC2 separated by disease status (e.g., healthy and T2D).</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># music2 deconvolution</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">1234</span><span class="op">)</span>
<span class="va">est.prop</span> <span class="op">=</span> <span class="fu"><a href="../../reference/music2_prop.html">music2_prop</a></span><span class="op">(</span>bulk.control.mtx <span class="op">=</span> <span class="va">bulk.control.mtx</span>, bulk.case.mtx <span class="op">=</span> <span class="va">bulk.case.mtx</span>, sc.sce <span class="op">=</span> <span class="va">seger.sce</span>, 
clusters <span class="op">=</span> <span class="st">'cellType'</span>, samples <span class="op">=</span> <span class="st">'sampleID'</span>, select.ct <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'acinar'</span>,<span class="st">'alpha'</span>,<span class="st">'beta'</span>,<span class="st">'delta'</span>,<span class="st">'ductal'</span>,<span class="st">'gamma'</span><span class="op">)</span>, n_resample<span class="op">=</span><span class="fl">20</span>, sample_prop<span class="op">=</span><span class="fl">0.5</span>,cutoff_c<span class="op">=</span><span class="fl">0.05</span>,cutoff_r<span class="op">=</span><span class="fl">0.01</span><span class="op">)</span><span class="op">$</span><span class="va">Est.prop</span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># plot estimated cell type proportions</span>
<span class="va">prop_all</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span><span class="op">(</span><span class="st">'proportion'</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">est.prop</span><span class="op">)</span>, <span class="st">'sampleID'</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">est.prop</span><span class="op">)</span>,times<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">est.prop</span><span class="op">)</span><span class="op">)</span>, <span class="st">'celltype'</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">est.prop</span><span class="op">)</span>, each<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">est.prop</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="va">prop_all</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">prop_all</span><span class="op">)</span>
<span class="va">prop_all</span><span class="op">$</span><span class="va">proportion</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">prop_all</span><span class="op">$</span><span class="va">proportion</span><span class="op">)</span><span class="op">)</span>
<span class="va">prop_all</span><span class="op">$</span><span class="va">group</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">prop_all</span><span class="op">$</span><span class="va">sampleID</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span>from<span class="op">=</span><span class="fl">1</span>, to<span class="op">=</span><span class="fl">100</span>, by<span class="op">=</span><span class="fl">1</span><span class="op">)</span>, <span class="st">'Healthy'</span>, <span class="st">'T2D'</span><span class="op">)</span>
<span class="va">cols</span> <span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"alpha"</span> <span class="op">=</span> <span class="st">"cadetblue2"</span>, <span class="st">"beta"</span> <span class="op">=</span> <span class="st">"lightsalmon1"</span>, <span class="st">"delta"</span> <span class="op">=</span> <span class="st">"palegreen2"</span>, <span class="st">"ductal"</span> <span class="op">=</span> <span class="st">"goldenrod1"</span>,
          <span class="st">"gamma"</span><span class="op">=</span><span class="st">"steelblue3"</span>, <span class="st">"acinar"</span> <span class="op">=</span> <span class="st">"plum2"</span><span class="op">)</span>
<span class="fu">ggplot</span><span class="op">(</span><span class="va">prop_all</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">celltype</span>, y<span class="op">=</span><span class="va">proportion</span>, color<span class="op">=</span><span class="va">celltype</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">xlab</span><span class="op">(</span><span class="st">''</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">geom_jitter</span><span class="op">(</span>width<span class="op">=</span><span class="fl">0.25</span>,alpha<span class="op">=</span><span class="fl">0.8</span><span class="op">)</span><span class="op">+</span><span class="fu">ylab</span><span class="op">(</span><span class="st">'Cell Type Proportions'</span><span class="op">)</span><span class="op">+</span><span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">stat_summary</span><span class="op">(</span>fun <span class="op">=</span> <span class="va">median</span>,
               geom <span class="op">=</span> <span class="st">"crossbar"</span>, width <span class="op">=</span> <span class="fl">0.5</span>,size<span class="op">=</span><span class="fl">0.5</span>,color<span class="op">=</span><span class="st">'gray36'</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">.</span><span class="op">~</span><span class="va">group</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">theme</span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>hjust <span class="op">=</span> <span class="fl">0.5</span>, size<span class="op">=</span><span class="fl">12</span><span class="op">)</span>,
        axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fl">12</span>,angle <span class="op">=</span> <span class="fl">45</span>,hjust<span class="op">=</span><span class="fl">1</span><span class="op">)</span>,
        axis.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fl">12</span><span class="op">)</span>,
        axis.title.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fl">12</span><span class="op">)</span>,
        axis.title.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fl">12</span><span class="op">)</span>,
        axis.line <span class="op">=</span> <span class="fu">element_line</span><span class="op">(</span>colour <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,
        strip.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,
        panel.grid.major <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
        panel.grid.minor <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
        panel.border <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
        panel.background <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
        legend.position <span class="op">=</span> <span class="st">'none'</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">scale_color_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">cols</span><span class="op">)</span></code></pre></div>
<p><img src="../images/cell_type_deconvolution_1-1.png"></p>
</div>
<div id="bechmark-evaluation" class="section level2">
<h2 class="hasAnchor">
<a href="#bechmark-evaluation" class="anchor"></a>Bechmark Evaluation</h2>
<p>We also deconvolved the benchmark bulk RNA-seq data using MuSiC <span class="citation">Wang et al. (2019)</span>, and evaluated the accuracy of both deconvolution methods by comparing the estimated cell type proportions obtained by <code>MuSiC2</code> and by <code>MuSiC</code> to the true proportions.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ALL</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span><span class="op">(</span><span class="va">prop_all</span><span class="op">[</span>,<span class="op">-</span><span class="fl">4</span><span class="op">]</span>,<span class="st">'Method'</span><span class="op">=</span><span class="st">'MuSiC2'</span><span class="op">)</span>
<span class="co"># MuSiC</span>
<span class="va">prop_music</span><span class="op">=</span><span class="fu"><a href="../../reference/music_prop.html">music_prop</a></span><span class="op">(</span>bulk.eset <span class="op">=</span> <span class="fu">exprs</span><span class="op">(</span><span class="va">benchmark.eset</span><span class="op">)</span>, sc.sce <span class="op">=</span> <span class="va">seger.sce</span>, 
                      clusters <span class="op">=</span> <span class="st">'cellType'</span>, samples <span class="op">=</span> <span class="st">'sampleID'</span>, 
                      select.ct <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'acinar'</span>,<span class="st">'alpha'</span>, <span class="st">'beta'</span>, <span class="st">'delta'</span>, <span class="st">'ductal'</span>,<span class="st">'gamma'</span><span class="op">)</span>, 
                      verbose <span class="op">=</span> <span class="cn">F</span><span class="op">)</span><span class="op">$</span><span class="va">Est.prop.weighted</span>
<span class="va">prop_all</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span><span class="op">(</span><span class="st">'proportion'</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">prop_music</span><span class="op">)</span>,<span class="st">'celltype'</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">prop_music</span><span class="op">)</span>, each<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">prop_music</span><span class="op">)</span><span class="op">)</span>, <span class="st">'sampleID'</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">prop_music</span><span class="op">)</span>,times<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">prop_music</span><span class="op">)</span><span class="op">)</span>,<span class="st">'Method'</span><span class="op">=</span><span class="st">'MuSiC'</span><span class="op">)</span>
<span class="va">prop_all</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">prop_all</span><span class="op">)</span>
<span class="va">prop_all</span><span class="op">$</span><span class="va">proportion</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">prop_all</span><span class="op">$</span><span class="va">proportion</span><span class="op">)</span><span class="op">)</span>
<span class="va">ALL</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="va">ALL</span>,<span class="va">prop_all</span><span class="op">)</span>

<span class="co"># true proportion</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="st">'./data/true_proportion.RData'</span><span class="op">)</span>
<span class="va">prop_all</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span><span class="op">(</span><span class="va">prop_all</span>,<span class="st">'Method'</span><span class="op">=</span><span class="st">'Truth'</span><span class="op">)</span>
<span class="va">ALL</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="va">ALL</span>,<span class="va">prop_all</span><span class="op">)</span>
<span class="va">ALL</span><span class="op">$</span><span class="va">group</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">ALL</span><span class="op">$</span><span class="va">sampleID</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span>from<span class="op">=</span><span class="fl">1</span>, to<span class="op">=</span><span class="fl">100</span>, by<span class="op">=</span><span class="fl">1</span><span class="op">)</span>, <span class="st">'Healthy'</span>, <span class="st">'T2D'</span><span class="op">)</span>
<span class="va">ALL</span><span class="op">$</span><span class="va">Method</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">ALL</span><span class="op">$</span><span class="va">Method</span>, levels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'Truth'</span>,<span class="st">'MuSiC2'</span>,<span class="st">'MuSiC'</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>Below we present the individual-level root mean square error (RMSE) across cell types for the two deconvolution methods separated by disease status (e.g., healthy and T2D) (Figure 3: left). As expected, because MuSiC2 only refines the gene list in the single cell reference when deconvolving bulk samples generated from clinical condition that differs from the single cell data, <code>MuSiC</code> and <code>MuSiC2</code> had exactly the same performance for healthy samples with estimation bias close to 0. For diseased samples, <code>MuSiC2</code> improved the estimation accuracy, highlighting the significance of gene selection for deconvolution. Especially for beta cells, MuSiC2 produced much more accurate cell type proportion estimates for diseased bulk samples than MuSiC, which suffered from severe underestimation (Figure 3: right).</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># root mean square error</span>
<span class="va">RMSE</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">proportion.x</span>, <span class="va">proportion.y</span><span class="op">)</span><span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">sqrt</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="op">(</span><span class="va">proportion.x</span><span class="op">-</span><span class="va">proportion.y</span><span class="op">)</span><span class="op">^</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span><span class="op">}</span>
<span class="va">prop_bias</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html">merge</a></span><span class="op">(</span><span class="va">ALL</span><span class="op">[</span><span class="va">ALL</span><span class="op">$</span><span class="va">Method</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'MuSiC'</span>,<span class="st">'MuSiC2'</span><span class="op">)</span>,<span class="op">]</span>, <span class="va">ALL</span><span class="op">[</span><span class="va">ALL</span><span class="op">$</span><span class="va">Method</span><span class="op">==</span><span class="st">'Truth'</span>,<span class="op">-</span><span class="fl">4</span><span class="op">]</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'sampleID'</span>,<span class="st">'celltype'</span>,<span class="st">'group'</span><span class="op">)</span><span class="op">)</span>
<span class="va">bias</span> <span class="op">=</span> <span class="va">prop_bias</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">Method</span>, <span class="va">sampleID</span>, <span class="va">group</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html">summarise</a></span><span class="op">(</span>rmse<span class="op">=</span><span class="fu">RMSE</span><span class="op">(</span><span class="va">proportion.x</span>, <span class="va">proportion.y</span><span class="op">)</span><span class="op">)</span>
<span class="va">p1</span><span class="op">=</span><span class="fu">ggplot</span><span class="op">(</span><span class="va">bias</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">Method</span>, y<span class="op">=</span><span class="va">rmse</span>, fill<span class="op">=</span><span class="va">group</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_boxplot</span><span class="op">(</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">ylab</span><span class="op">(</span><span class="st">'RMSE'</span><span class="op">)</span><span class="op">+</span><span class="fu">xlab</span><span class="op">(</span><span class="st">''</span><span class="op">)</span><span class="op">+</span><span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">theme</span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fl">12</span><span class="op">)</span>,
        axis.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fl">12</span><span class="op">)</span>,
        axis.title.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fl">12</span><span class="op">)</span>,
        axis.title.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fl">12</span><span class="op">)</span>,
        axis.line <span class="op">=</span> <span class="fu">element_line</span><span class="op">(</span>colour <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,
        strip.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,
        panel.grid.major <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
        panel.grid.minor <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
        panel.border <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
        panel.background <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
        legend.position <span class="op">=</span> <span class="st">'none'</span>,
        legend.title <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span><span class="op">)</span>

<span class="co"># beta cell</span>
<span class="va">p2</span><span class="op">=</span><span class="fu">ggplot</span><span class="op">(</span><span class="va">ALL</span><span class="op">[</span><span class="va">ALL</span><span class="op">$</span><span class="va">celltype</span><span class="op">==</span><span class="st">'beta'</span>,<span class="op">]</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">Method</span>, y<span class="op">=</span><span class="va">proportion</span>, fill<span class="op">=</span><span class="va">group</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_boxplot</span><span class="op">(</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">ylab</span><span class="op">(</span><span class="st">'Beta Cell Proportions'</span><span class="op">)</span><span class="op">+</span><span class="fu">xlab</span><span class="op">(</span><span class="st">''</span><span class="op">)</span><span class="op">+</span><span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">theme</span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fl">12</span><span class="op">)</span>,
        axis.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fl">12</span><span class="op">)</span>,
        axis.title.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fl">12</span><span class="op">)</span>,
        axis.title.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fl">12</span><span class="op">)</span>,
        axis.line <span class="op">=</span> <span class="fu">element_line</span><span class="op">(</span>colour <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,
        strip.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,
        panel.grid.major <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
        panel.grid.minor <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
        panel.border <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
        panel.background <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
        legend.position <span class="op">=</span> <span class="st">'right'</span>,
        legend.title <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span><span class="op">)</span>

<span class="fu">grid.arrange</span><span class="op">(</span><span class="va">p1</span>,<span class="va">p2</span>,widths<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">3</span>,<span class="fl">4</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="../images/compare-1.png"></p>
</div>
</div>
<div id="reference" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#reference" class="anchor"></a>Reference</h1>
<div id="refs" class="references">
<div id="ref-segerstolpe2016single">
<p>Segerstolpe, Åsa, Athanasia Palasantza, Pernilla Eliasson, Eva-Marie Andersson, Anne-Christine Andréasson, Xiaoyan Sun, Simone Picelli, et al. 2016. “Single-Cell Transcriptome Profiling of Human Pancreatic Islets in Health and Type 2 Diabetes.” <em>Cell Metabolism</em> 24 (4): 593–607.</p>
</div>
<div id="ref-wang2019bulk">
<p>Wang, Xuran, Jihwan Park, Katalin Susztak, Nancy R Zhang, and Mingyao Li. 2019. “Bulk Tissue Cell Type Deconvolution with Multi-Subject Single-Cell Expression Reference.” <em>Nature Communications</em> 10 (1): 1–9.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Xuran Wang, Jiaxin Fan.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
