<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>MuSiC2_Deconvolution — music2_prop • MuSiC</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="MuSiC2_Deconvolution — music2_prop"><meta property="og:description" content="This function is used to deconvolve bulk RNA-seq data using single-cell reference generated under a different condition."><meta property="og:image" content="http://xuramw.github.io/MuSiC/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MuSiC</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/MuSiC.html">Tutorial</a>
</li>
<li>
  <a href="../articles/pages/MuSiC2.html">MuSiC2</a>
</li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li>
  <a href="../articles/pages/data.html">Data</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/xuranw/MuSiC" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>MuSiC2_Deconvolution</h1>
    <small class="dont-index">Source: <a href="https://github.com/xuranw/MuSiC/blob/HEAD/R/music2.R" class="external-link"><code>R/music2.R</code></a></small>
    <div class="hidden name"><code>music2_prop.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function is used to deconvolve bulk RNA-seq data using single-cell reference generated under a different condition.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">music2_prop</span><span class="op">(</span></span>
<span>  <span class="va">bulk.control.mtx</span>,</span>
<span>  <span class="va">bulk.case.mtx</span>,</span>
<span>  <span class="va">sc.sce</span>,</span>
<span>  <span class="va">clusters</span>,</span>
<span>  <span class="va">samples</span>,</span>
<span>  <span class="va">select.ct</span>,</span>
<span>  expr_low <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  prop_r <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  eps_c <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  eps_r <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>  n_resample <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  sample_prop <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  cutoff_expr <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  cutoff_c <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  cutoff_r <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>  maxiter <span class="op">=</span> <span class="fl">200</span>,</span>
<span>  markers <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  cell_size <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  ct.cov <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  centered <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  normalize <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>bulk.control.mtx</dt>
<dd><p>Matrix of expression for bulk data, control group</p></dd>


<dt>bulk.case.mtx</dt>
<dd><p>Matrix of expression for bulk data, case group</p></dd>


<dt>sc.sce</dt>
<dd><p>SingleCellExperiment for single cell data</p></dd>


<dt>clusters</dt>
<dd><p>character, the phenoData of single cell dataset used as clusters;</p></dd>


<dt>samples</dt>
<dd><p>character, the phenoData of single cell dataset used as samples;</p></dd>


<dt>select.ct</dt>
<dd><p>vector of cell types, default as NULL. If NULL, then use all cell types provided by the single cell dataset;</p></dd>


<dt>expr_low</dt>
<dd><p>numeric, cutoff for defining lowly expressed genes in bulk data. Genes with mean expression across samples in bulk data &lt; expr_low will be excluded from cell-type-specific DE gene detection. Default is 20;</p></dd>


<dt>prop_r</dt>
<dd><p>numeric, cutoff for defining rare cell types. Cell types with mean proportion across samples in bulk data &lt; prop_r will be characterized as rare cell types. Otherwise, will be characterized as common cell types. Default is 0.1;</p></dd>


<dt>eps_c</dt>
<dd><p>numeric, convergence cutoff for common cell types. The cell type proportion estimate is converged if absolute relative change of proportion estimates for the current iteration against the previous iteration &lt; eps_c. Default is 0.05;</p></dd>


<dt>eps_r</dt>
<dd><p>numeric, convergence cutoff for rare cell types. The cell type proportion estimate is converged if absolute change of proportion estimates for the current iteration against the previous iteration &lt; eps_r. Default is 0.01;</p></dd>


<dt>n_resample</dt>
<dd><p>numeric, number of resamples used for detecting cell-type-specific DE genes. Default is 20;</p></dd>


<dt>sample_prop</dt>
<dd><p>numeric, proportion of samples to be randomly sampled without replacement under each condition at each resampling. Default is 0.5;</p></dd>


<dt>cutoff_expr</dt>
<dd><p>numeric, cutoff for defining lowly expressed genes over resamples. Genes with average cell-type-specific expression calculated over all resamples in the lower cutoff_expr quantile are excluded from cell-type-specific DE gene detection. Default is 0.05;</p></dd>


<dt>cutoff_c</dt>
<dd><p>numeric, cutoff for defining cell-type-specific DE genes for common cell types. Genes with the value of statistic, defined as the absolute value of the ratio of the mean and standard deviation of the log fold change over all resamples, in the upper cutoff_c quantile are considered as cell-type-specific DE genes. Default is 0.05;</p></dd>


<dt>cutoff_r</dt>
<dd><p>numeric, cutoff for defining cell-type-specific DE genes for rare cell types. Genes with the value of statistic, defined as the absolute value of the ratio of the mean and standard deviation of the log fold change over all resamples, in the upper cutoff_r quantile are considered as cell-type-specific DE genes. Default is 0.01;</p></dd>


<dt>maxiter</dt>
<dd><p>numeric, maximum number of iterations. Default is 200;</p></dd>


<dt>markers</dt>
<dd><p>vector or list of gene names. Default as NULL, i.e., use all genes that provided by both bulk and single cell datasets;</p></dd>


<dt>cell_size</dt>
<dd><p>data.frame of cell sizes. 1st column contains the names of cell types, 2nd column has the cell sizes per cell type. Default as NULL. If NULL, then estimate cell size from data;</p></dd>


<dt>ct.cov</dt>
<dd><p>logical. If TRUE, use the covariance across cell types;</p></dd>


<dt>centered</dt>
<dd><p>logic, subtract avg of Y and D;</p></dd>


<dt>normalize</dt>
<dd><p>logic, divide Y and D by their standard deviation;</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>If MuSiC2 converges, return:</p><ul><li><p>Est.prop: matrix, cell type proportion estimates.</p></li>
<li><p>convergence: logical, whether MuSiC2 converged or not.</p></li>
<li><p>n.iter: numeric, number of iterations.</p></li>
<li><p>DE.genes: vector, cell-type-specific DE genes being removed.</p></li>
</ul><p>Or if MuSiC2 does not converge, return:</p>
<p></p>
<ul><li><p>Est.prop: matrix, cell type proportion estimates.</p></li>
<li><p>convergence: logical, whether MuSiC2 converged or not.</p></li>
<li><p>id.not.converge: vector, sample ids that failed to converge.</p></li>
</ul></div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="music_prop.html">music_prop</a></code></p></div>
    </div>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Xuran Wang, Jiaxin Fan.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

